### PR TITLE
chore: Add KeyId computation while importing secret in ockam_vault

### PR DESCRIPTION
**Pull request by devalain**
### Proposed change
Simply add a private function to `SoftwareVault` that would be called from `secret_import`.

### CODE REVIEW

**Comment by mrinalwadhwa** [comment]
@devalain Thank you for sending the PR and picking Ockam to make you first open source contribution ❤️ 🙏 

While we wait for @SanjoDeundiak's review, could you accept our CLA ...

Read the CLA at
https://www.ockam.io/learn/how-to-guides/contributing/cla/

To accept the CLA please send a different PR to our contributors repo indicating that you accept the CLA by adding your Git/Github details in a row at the end of the CONTRIBUTORS.csv file:
https://github.com/ockam-network/contributors/blob/master/CONTRIBUTORS.csv

Looking forward to merging your first contribution!!

File: implementations/rust/ockam/ockam_vault/src/secret_impl.rs
Comment (SanjoDeundiak) [inline implementations/rust/ockam/ockam_vault/src/secret_impl.rs:41]: Can we explicitly enumerate all members here, so that whenever we add new secret type we won't be able to compile library until all affected places are updated?

**Review by SanjoDeundiak — CHANGES_REQUESTED** [review]
Hey @devalain !  Thanks for your contribution. There are few things we need to fix before merging your PR
1. Could you please [update](https://docs.github.com/en/github/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message#rewriting-the-most-recent-commit-message) your commit message to match our [guide](https://www.ockam.io/learn/how-to-guides/contributing/CONTRIBUTING/#commit-messages)?
2. Could you please add simple tests to check your fix? Easiest case would be checking that after both key generation and key import you can extract public key using secret_public_key_get, get key id using compute_key_id_for_public_key and obtain the secret from vault using get_secret_by_key_id?
3. Ideal case is to also add simple test vector with at least 1 predefined BLS and 1 predefined curve25519 key for which key id is computed and compared to predefined values. That's optional
4. After you add tests from p.2 you'll notice that secret_import still always uses None as key id value.

**Comment by devalain** [comment]
Hey @SanjoDeundiak, I added some tests as you suggested in p.1 and I will do the p.2. I did not notice p.3, could you give me more detail about it please ?

**Comment by SanjoDeundiak** [comment]
@devalain I think it's github's UI tricked me into thinking that you replaced key id computation in secret_generate with function call and did not add it to secret_import, but it's vice versa. What do you think about replacing key id computation in secret_generate with a call of your newly added function?
Other than that looks great to me. Waiting for the last test to be added.
P.S. Could you please also run cargo fmt on your code?

**Comment by devalain** [comment]
> What do you think about replacing key id computation in secret_generate with a call of your newly added function?
I thought I would not because then a call to `secret_generate` would compute public keys 2 times. Maybe there is a good reason I didn't think of ?

> Other than that looks great to me. Waiting for the last test to be added.
> P.S. Could you please also run cargo fmt on your code?
Just did it :)

**Comment by SanjoDeundiak** [comment]
@devalain I think secret_generate computes public key only for further computation of key id. So there shouldn't be double computation

File: implementations/rust/ockam/ockam_vault/src/secret_impl.rs
Comment (SanjoDeundiak) [inline implementations/rust/ockam/ockam_vault/src/secret_impl.rs:250]: I don't think these are valid keys, so as we add key validation during import this test will fail. Let's generate some random keys for this test.
P.S. I hope Maxwell pardons me

Comment (SanjoDeundiak) [reply implementations/rust/ockam/ockam_vault/src/secret_impl.rs]: > Ok, I put some random (and boring ;)) keys.
> 
> Joke aside, how do I validate secret keys in ockam ?

I was trying to say that not every 32 byte binary is a valid curve25519 secret key, and I think this is true for any asymmetric cryptography (not 100% sure). Depending on curve there are some arithmetic checks needed to be done to ensure key is valid. We currently don't have them.

**Comment by SanjoDeundiak** [comment]
@devalain merging as #1610

File: implementations/rust/ockam/ockam_vault/src/secret_impl.rs
Comment (devalain) [reply implementations/rust/ockam/ockam_vault/src/secret_impl.rs]: > We currently don't have them.

Wouldn't that be an interesting feature to add to ockam ? I'm not sure in which crate exactly.

Comment (mrinalwadhwa) [reply implementations/rust/ockam/ockam_vault/src/secret_impl.rs]: @devalain yes that kind of validation would be a great addition.

Comment (SanjoDeundiak) [reply implementations/rust/ockam/ockam_vault/src/secret_impl.rs]: > > We currently don't have them.
> 
> Wouldn't that be an interesting feature to add to ockam ? I'm not sure in which crate exactly.

I think it goes straight to to secret_import in ockam_vault

**Pull request closed without merging**
